great_expectations/rule_based_profiler/parameter_builder/mean_unexpected_metric_multi_batch_parameter_builder.py

```diff
@@ -0,0 +1,191 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+
+from great_expectations.core.batch import Batch, BatchRequest, RuntimeBatchRequest
+from great_expectations.rule_based_profiler.helpers.util import (
+    get_parameter_value_and_validate_return_type,
+)
+from great_expectations.rule_based_profiler.parameter_builder import (
+    MetricMultiBatchParameterBuilder,
+)
+from great_expectations.rule_based_profiler.parameter_builder.parameter_builder import (
+    MetricValues,
+)
+from great_expectations.rule_based_profiler.types import (
+    Domain,
+    ParameterContainer,
+    ParameterNode,
+)
+
+
+class MeanUnexpectedMapMetricMultiBatchParameterBuilder(
+    MetricMultiBatchParameterBuilder
+):
+    """
+    Compute mean unexpected count ratio (as a fraction) of a specified map-style metric across all specified batches.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        map_metric_name: str,
+        total_count_parameter_builder_name: str,
+        null_count_parameter_builder_name: Optional[str] = None,
+        metric_domain_kwargs: Optional[Union[str, dict]] = None,
+        metric_value_kwargs: Optional[Union[str, dict]] = None,
+        batch_list: Optional[List[Batch]] = None,
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+        json_serialize: Union[str, bool] = True,
```

Comment:
Why can this be a string as well?

Reply:
@cdkini Because of the `$parameter` syntax, potentially used.